src/App.tsx

```diff
@@ -15,48 +17,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'alphabet',
+  Length = 'length',
+  None = '',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: SortType,
+  reverseField = false,
+) {
+  const visibleGoods = [...goods].map((good, index) => (
+    {
+      name: good,
+      id: (index + 1),
+      long: good.length,
+    }
+  ));
+
+  if (sortField) {
+    visibleGoods.sort((a, b): number => {
+      switch (sortField) {
+        case SortType.Alphabet:
+          return a.name.localeCompare(b.name);
+
+        case SortType.Length:
+          return a.long - b.long;
```

Comment:
Why complicate the code so much? Here it is enough to simply compare the length
```suggestion
          return a.length - b.length;
```